--- rootfs.original/home/lava/klipper/klippy/extras/filament_protocol.py
+++ rootfs/home/lava/klipper/klippy/extras/filament_protocol.py
@@ -36,6 +36,7 @@
     'OFFICIAL': False,
     'CARD_UID': 0,
     'CARD_TYPE': '',
+    'CARD_EVENT_TIME': 0,
 }
 
 # Filament main type
--- rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py
+++ rootfs/home/lava/klipper/klippy/extras/filament_detect.py
@@ -104,6 +104,7 @@
 
 
     def _filament_info_update(self, channel, info, is_clear=False):
+        info['CARD_EVENT_TIME'] = self.reactor.monotonic()
         self._filament_info[channel] = info
 
         # notify
@@ -251,6 +252,7 @@
                 raise web_request.error("unsupported fields: %s" % (', '.join(sorted(params.keys()))))
 
             filament_info['OFFICIAL'] = has_params
+            filament_info['CARD_EVENT_TIME'] = self.reactor.monotonic()
             self._state[channel] = FILAMENT_DT_STATE_IDLE
 
             ptc = self.printer.lookup_object('print_task_config', None)
